runtime/vm/swalk.c

```diff
@@ -123,6 +123,9 @@ UDATA  walkStackFrames(J9VMThread *currentThread, J9StackWalkState *walkState)
 	if (currentThread != NULL) {
 		oldState = currentThread->activeWalkState;
 		currentThread->activeWalkState = walkState;
+		walkState->javaVM = currentThread->javaVM;
+	} else {
+		walkState->javaVM = walkState->walkThread->javaVM;
```

Comment:
You can do this unconditionally and remove the other one. walkThread must always be valid.